tests/data/claypit/tests/advanced_coverage.suite.yml

```diff
@@ -0,0 +1,8 @@
+class_name: AdvancedCoverageTester
+modules:
+    enabled: [Asserts]
+coverage:
+    enabled: true
+    advanced:
```

Comment:
I don't think "advanced" have any meaning here. 
code coverge settings are already as advanced as possible 😄 

so please add those settings to `coverage` level and **document them** at CodeCoverage guide

Reply:
or maybe if I understood it wrong, please find a better name instead of "advanced"

Reply:
They are settings that are proxied directly to the underlying library; that is what makes them advanced.
Since we also have settings for the Codeception module I did not want to put them in the same namespace; it doesn't improve readability.
Currently we have these settings:
```php
protected $defaultSettings = [
        'enabled'        => false,
        'remote'         => false,
        'local'          => false,
        'xdebug_session' => 'codeception',
        'remote_config'  => null,
        'show_uncovered' => false,
        'c3_url'         => null,
        'work_dir'       => null,
        'cookie_domain'  => null,
    ];
```

They are not the same, or even related to, the settings for the `php-code-coverage` library. That is why they do not live in the same namespace.

I'm not gonna bother discussing the name here since I honestly don't care about it :), if you have any other suggestion I'd be happy to change it.

Reply:
> so please add those settings to `coverage` level and document them at CodeCoverage guide

This is something I explicitly will not do. It would only serve as a duplicate of the underlying library documentation and will quickly grow out of date.
Instead I have referred to the source code of the library (since there is no documentation for all its configuration options). -- This leads us back to the original name: `advanced`.

Basically this is a way for you to go deep into the inner workings and change stuff, it is advanced because it requires reading the library source code and because troubles from using it will likely not be supported by the Codeception community.

Reply:
@SamMousa 
thanks, now I got the point. How about naming it "phpCodeCoverage" - which is just how you explained things.

```yaml
coverage:
  phpCodeCoverage:
    # here go advanced settings
```
?

Reply:
Yeah fine with me. Will change it after the weekend